test/integration/AuthenticatedFileResourceStore.test.ts

```diff
@@ -0,0 +1,175 @@
+import { promises as fs } from 'fs';
+import { join } from 'path';
+import * as url from 'url';
+import { MockResponse } from 'node-mocks-http';
+import { HttpHandler, ResourceStore } from '../../index';
+import { AuthenticatedFileResourceStoreConfig } from '../configs/AuthenticatedFileResourceStoreConfig';
+import { RUNTIMECONFIG } from '../configs/Util';
+import { AclTestHelper } from '../util/TestHelpers';
+import { call, callFile } from '../util/Util';
+
+describe('A server using a AuthenticatedFileResourceStore', (): void => {
+  let config: AuthenticatedFileResourceStoreConfig;
+  let handler: HttpHandler;
+  let store: ResourceStore;
+  let aclHelper: AclTestHelper;
+
+  beforeAll(
+    async(): Promise<void> => {
+      config = new AuthenticatedFileResourceStoreConfig();
+      handler = config.getHttpHandler();
+      ({ store } = config);
+      aclHelper = new AclTestHelper(store, RUNTIMECONFIG.base);
+
+      const root = RUNTIMECONFIG.rootFilepath;
+
+      // Make sure the root directory exists
+      await fs.mkdir(join(__dirname, '../../uploads'), { recursive: true });
+      await fs.copyFile(join(__dirname, '../assets/permanent.txt'), join(__dirname, `../../${root}/permanent.txt`));
```

Comment:
It's a bit weird that `root` is not really the root. Might be cleaner to just make sure that `root` points to the exact root folder from the beginning.